commerce/domain/tax/cart/calculator/classes/TaxCartCalculatorSample.cls

```diff
@@ -34,13 +40,20 @@ public class TaxCartCalculatorSample extends CartExtension.TaxCartCalculator {
       // The cartItemCollection contains both products and shipping cart items.
       Map<String, CartExtension.CartItem> cartItemById = new Map<String, CartExtension.CartItem>();
       Map<String, CartExtension.CartItem> shippingItemById = new Map<String, CartExtension.CartItem>();
-      for (Integer i = (cartItemCollection.size() - 1); i >= 0; i--) {
-        if (cartItemCollection.get(i).getType() == CartExtension.SalesItemTypeEnum.PRODUCT) {
-          cartItemById.put(cartItemCollection.get(i).getId(), cartItemCollection.get(i));
-        } else if (cartItemCollection.get(i).getType() == CartExtension.SalesItemTypeEnum.CHARGE) {
+
+      Iterator<CartExtension.CartItem> cartItemCollectionIterator = cartItemCollection.iterator();
+
+      while (cartItemCollectionIterator.hasNext()) {
+        CartExtension.CartItem cartItem = cartItemCollectionIterator.next();
+
+        if (cartItem.getType() == CartExtension.SalesItemTypeEnum.PRODUCT) {
+          if (cartItem.getId() != null) {
```

Comment:
Hm...but in this case you would not calculate taxes for the CartItem that has just been added to cart (does not have an Id yet). Is this correct? Could you please explain?